eclipse-che/che-operator
# Sync the certificate from che-git-self-signed-cert to user namespaces

### What does this PR do?
`$TITLE` so that it can be mounted into devworkspaces.

### What issues does this PR fix or reference?
eclipse/che#20528

This depends on https://github.com/devfile/devworkspace-operator/pull/625

### How to test this PR?
1. Create a config map called `che-git-self-signed-cert` in the namespace where `CheCluster` resource is stored with contents like this:
```yaml
kind: ConfigMap
apiVersion: v1
metadata:
  name: che-git-self-signed-cert
data:
  githost: somewhere.else
  ca.crt: <<contents of some certificate or any string really for this demonstration purposes >>
```
2. Set the `spec.server.gitSelfSignedCert` property of the `CheCluster` CR to `true`.
3. Observe that in all namespaces marked with the `che.eclipse.org/workspace-namespace-owner-uid` annotation, there is a new configmap created with a name based on the name and namespace of the CheCluster resource in the form:
```yaml
kind: ConfigMap
apiVersion: v1
metadata:
  name: <<CheCluster-cr-name>>-<<CheCluster-cr-namespace>>-git-tls-creds
  namespace: <<the-workspace-namespace>>
  labels:
    controller.devfile.io/git-tls-credential: "true"
    controller.devfile.io/watch-configmap: "true"
data:
  host: somewhere.else
  certificate: <<contents of some certificate or any string really for this demonstration purposes >>
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
